content/knowledge/consensus_algorithms.md

```diff
@@ -0,0 +1,367 @@
+At this point it is safe to act on the value - meaning to include it in its local ledger.
+In Stellar this is also called externalizing the value.
+Once a ballot is confirmed committed, a node can be sure that every other well-behaved node has externalized, or inevitably will externalize, the same value.
+
+Using the FBA voting the Stellar network is able to achieve liveness and safety for the network, as well as to FBA consensus is the most scalable solution, first implemented by Ripple (XRP) and then improved upon by Stellar (XLM). The consensus requires a special network of nodes to be known and verified ahead of time before put into use.
+
+TODO: finish with conclusion
+
+#### Ripple
+
+[Ripple consensus](https://ripple.com/files/ripple_consensus_whitepaper.pdf) is an example of a voting consensus that has some number of voting nodes for each round, but the total number of voters in the system is unknown.
+
+Each node that receives answers from its queried voters has knowledge of their number and can calculate majority of them or a certain percentage of them, but their number does not correspond in anyway with all voters in the network.
+
+The consensus has a few components that comprise its notions for reaching an agreement: the state in which nodes in the network reach agreement.
+
+First component is the server.
+A server is any entity running the Ripple Server software which participats in the consensus protocol.
+The second component is the ledger and it is a record of each user account representing the _ground truth_ of the network.
+The notions of a server and a ledger are identical with those in rest of the described protocols in this documentation, as well as in blockchain networks in general.
+
+Last-Closed Ledger is the most recent ledger that has been ratified by the consensus process and thus represents the current state of the network, as opposed to open ledger, which is the current operating status of the ledger of each node.
+Transactions sent from the network users become part of the open ledger for each node, but of course they are considered final only after they are included in the last-closed ledger after they pass through the consensus process.
+
+A Unique Node List (UNL) is a unique list of server nodes that each of the servers maintains.
+Each server queries these nodes when executing consensus and only the votes from UNL members are considered.
+Consequently the UNL represents a subnet of the Ripple network, and it is considered trustful only as a whole.
+This means the set is expected to not collude in an attempt to defraud the network, but each single member of it is not implicitly condisered trustful.
+
+Any of the servers can broadcast transactions to be included in the consensus process and every server attempts to include every valid transaction when a new consensus round starts.
```

Comment:
This place has been rephrased here https://github.com/optakt/optakt.github.io/pull/21/commits/ca80e6c74bee64b8ce712ca515bb205ee645d971#diff-4bf8323c6426a93da1ce22be1b8528779ef6b61fdacef3e0d065f1e1dd47bea5R30
in order to avoid too much details